### PR TITLE
GH-1675: Validation on @KafkaHandler Methods

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves https://github.com/spring-projects/spring-kafka/issues/1675

Add support for payload validation with `@KafkaHandler` methods.

**cherry-pick to 2.6.x, 2.5.x**

### CODE REVIEW

File: spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java [hunk 1]
```diff
@@ -114,13 +129,35 @@ public DelegatingInvocableHandler(List<InvocableHandlerMethod> handlers,
...
+	/**
+	 * Construct an instance with the supplied handlers for the bean.
+	 * @param handlers the handlers.
+	 * @param defaultHandler the default handler.
+	 * @param bean the bean.
+	 * @param beanExpressionResolver the resolver.
+	 * @param beanExpressionContext the context.
+	 * @param beanFactory the bean factory.
+	 * @param validator the validator.
+	 * @since 2.1.11
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java:154]: `2.5.11`?

Why then you say in the PR description that we need to back-port it back to `2.3.x`?

**Review by artembilan — CHANGES_REQUESTED** [review]
OK. I can live with this solution - only need to agree with `@since` and all the back-port versions.

Thanks

File: spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java (hunk 1 above)
Comment (garyrussell) [reply spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/DelegatingInvocableHandler.java]: Fixed and updated PR comment.

**Comment by artembilan** [comment]
... and cherry-picked to `2.6.x` & `2.5.x` after fixing conflicts in `What's New`.

**Pull request merged**